gensim/models/keyedvectors.py

```diff
@@ -338,6 +345,129 @@ def most_similar(self, positive=[], negative=[], topn=10, restrict_vocab=None, i
         result = [(self.index2word[sim], float(dists[sim])) for sim in best if sim not in all_words]
         return result[:topn]
 
+    def most_similar_among(self, positive=[], negative=[],
+                            topn=10, words_list=None, indexer=None,
+                            suppress_warnings=False):
+        """
+        Find the top-N most similar words among words_list to given words.
+
+        Positive words contribute positively towards the similarity,
+        negative words negatively.
+
+        Please refer to docs of most_similar function.
+
+        If topn is False, most_similar returns the vector of similarity scores
+        for all words in vocabulary of model, restriced by the supplied words_list.
+
+        'words_list' should be a list/set of words. The returned word similarities
+        will only contain similarity scores for those words that are in words_list
+        (and in trained vocabulary).
+
+        If some words in words_list are not in vocabulary then a warning is
+        issued to the user.
+
+        Warnings can be supressed by setting the suppress_warnings flag.
+
+        Example::
+
+          >>> trained_model.most_similar_among(positive=['man'], topn=1,
+                                                words_list=['woman','random_word'])
+          [('woman', 0.75882536)]
+
+        """
+
+        if isinstance(words_list, int):
```

Comment:
Please use a single check and a single  `raise ValueError`. 
The `most_similar` function doesn't take a list of `int`s so it should not be mentioned here